spec/datadog/core/crashtracking/component_spec.rb

```diff
@@ -1,9 +1,43 @@
 require 'spec_helper'
 require 'datadog/core/crashtracking/component'
+require 'datadog/core/crashtracking/crashtracking_runtime_stacks'
+
+# Test helper toexpose runtime stack callback state used only inside this spec
+module Datadog
+  module Core
+    module Crashtracking
+      class Component
+        def runtime_callback_registered?
+          return false if Crashtracking::RUNTIME_STACKS_FAILURE
+          return false unless Crashtracking.const_defined?(:RuntimeStacks, false)
+
+          Crashtracking::RuntimeStacks._native_is_runtime_callback_registered
+        rescue => e
+          logger.debug("Runtime stack callback status check not available: #{e.message}")
+          false
+        end
+      end
+    end
+  end
+end
 
 require 'webrick'
 require 'fiddle'
 
+class ComponentSpecCrashStackHelper
+  def top_level_ruby_method
+    ruby_method_with_c_calls
+  end
+
+  def ruby_method_with_c_calls
+    'hello world'.gsub("world") do |_match|
+      {a: 1, b: 2}.each do |key, _value|
+        Fiddle.free(42) if key == :a
```

Comment:
Minor: Small simplification ;)

```suggestion
        Fiddle.free(42)
```